[statsdreceiver] add datadog_summary observer

diurnalist opened the pull request:
**Description:**

this adds a new observer that produces a "summary" of histograms similar to how datadog does it. this is not a distinct metric type in datadog, but instead is a convention where a sum and multiple gauges are emitted for the incoming histogram. the sum tracks the total number of samples and the gauges track mean, median, sum, and p95 aggregations over the sample set.

the primary use-case for this observer is admittedly niche; it is designed for cases where you are substituting the OpenTelemetry Collector in place of the Datadog Agent for DogStatsD ingest, but don't want to break existing dependencies on existing metrics that aggregate over histograms.

**Testing:** added a new test for both sampled and unsampled summaries.

**Documentation:** updated the README.

diurnalist committed:
[statsdreceiver] add datadog_summary observer

this adds a new observer that produces a "summary" of histograms
similar to how datadog does it. this is not a distinct metric
type in datadog, but instead is a convention where a sum and multiple
gauges are emitted for the incoming histogram. the sum tracks the
total number of samples and the gauges track mean, median, sum, and
p95 aggregations over the sample set.

the primary use-case for this observer is admittedly niche; it is
designed for cases where you are substituting the OpenTelemetry
Collector in place of the Datadog Agent for DogStatsD ingest, but
don't want to break existing dependencies on existing metrics that
aggregate over histograms.

Commit:
Update doc.go of filelogreceiver (#29100)

songy23 committed:
[connector/datadog] Set MutatesData to true (#29114)

**Description:** 
Mark datadogconnector as `MutatesData` to prevent data race

**Link to tracking Issue:**
Fixes
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/29111

axw committed:
cmd/telemetrygen: add HTTP export for logs (#29078)

**Description:**

Closes
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18867

**Testing:**

Ran opentelemetry-collector locally with debug exporter, then used
telemetrygen with `--otlp-http` with and without `--otlp-insecure`.

**Documentation:** None

mx-psi committed:
[processor/resourcedetection] Add feature gate to change family and model to string (#29152)

**Description:** 

Add feature gate to change family and CPU model to string, to adapt to
open-telemetry/semantic-conventions/issues/495.

**Link to tracking Issue:** Fixes #29025

**Testing:** Tested with the sample configuration on #26533.

nikore commented on receiver/statsdreceiver/internal/protocol/metric_translator.go:
```diff
@@ -92,6 +93,79 @@ func buildSummaryMetric(desc statsDMetricDescription, summary summaryMetric, sta
+
+	mCountDp := mCountSum.DataPoints().AppendEmpty()
+	// Note: count is rounded here, see note in counterValue().
+	mCountDp.SetIntValue(int64(count))
```
I belive we need to use `math.Round` here as the conversions [spec](https://go.dev/ref/spec#Conversions) under "Conversions between numeric types"  in golang say:

```
When converting a [floating-point number](https://go.dev/ref/spec#Numeric_types) to an integer, the fraction is discarded (truncation towards zero).
```

nikore reviewed (approved):
Overall this seems fine to me, will be interesting to see if this gets accepted upstream.

Dennis40816 committed:
[pkg/ottl] Add BoolGetter and IsBool converter (#27900)

**Description:**
This PR focuses on adding a new feature to the pkg/ottl package—namely,
the `IsBool` converter, along with its prerequisites, `BoolGetter` and
`BoolLikeGetter`. The series of commits implement the core logic, unit
tests, and documentation to ensure that the `IsBool` converter is
seamlessly integrated into the OTTL framework. By using this feature,
users can conveniently determine whether a given value is a boolean or
not.

**Link to tracking Issue:** 
#27897

**Testing:**

- A comprehensive set of unit tests were added to validate the
functionality of the `IsBool` converter. The tests cover native boolean
values, `pcommon.ValueTypeBool` objects, and incorrect types such as
integers and slices.
- Tests for `BoolGetter` and `BoolLikeGetter` were also included to
verify their behavior and error handling mechanisms. Detailed test
implementations can be found in
[pkg/ottl/expression_test.go](https://github.com/Dennis40816/opentelemetry-collector-contrib/blob/0d1c2c0216f5647404573a6cfe66ebc0081a4167/pkg/ottl/expression_test.go#L1452).

**Documentation:**
- Updated README.md in the `pkg/ottl/ottlfuncs` directory to include
detailed descriptions, usage guidelines, and examples for the `IsBool`
converter.
- The README file also reflects the addition of `BoolGetter` and
`BoolLikeGetter` to the list of supported types for single-value
parameters in OTTL functions.

Commit:
[chore] update collector core (#29122)

Also updated the references to NewNoopTracerProvider.

Fixes #29158

---------

Signed-off-by: Alex Boten <aboten@lightstep.com>

opentelemetrybot committed:
[chore] dependabot updates Tue Nov 14 02:40:36 UTC 2023 (#29262)

Bump actions/github-script from 6 to 7
Bump github.com/DataDog/opentelemetry-mapping-go/pkg/inframetadata from
0.8.0 to 0.8.1 in /exporter/datadogexporter
Bump github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/logs from
0.8.0 to 0.8.1 in /exporter/datadogexporter
Bump github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/metrics from
0.8.0 to 0.8.1 in /exporter/datadogexporter
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/exporter/awscloudwatchlogsexporter
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/exporter/awsemfexporter
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/exporter/awsxrayexporter
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/exporter/datadogexporter
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/extension/observer/ecsobserver
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/internal/aws/cwlogs
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/internal/aws/proxy
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/internal/aws/xray
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/internal/metadataproviders
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/processor/resourcedetectionprocessor
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/receiver/awscontainerinsightreceiver
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/receiver/awsecscontainermetricsreceiver
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/receiver/awsxrayreceiver
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.9 in
/internal/aws/awsutil
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.9 in
/internal/aws/k8s
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.9 in
/internal/aws/xray/testdata/sampleapp
Bump github.com/aws/aws-sdk-go-v2 from 1.22.1 to 1.22.2 in
/exporter/awskinesisexporter
Bump github.com/aws/aws-sdk-go-v2 from 1.22.1 to 1.22.2 in
/extension/sigv4authextension
Bump github.com/aws/aws-sdk-go-v2/config from 1.22.2 to 1.24.0 in
/exporter/awskinesisexporter
Bump github.com/aws/aws-sdk-go-v2/config from 1.22.2 to 1.24.0 in
/extension/sigv4authextension
Bump github.com/aws/aws-sdk-go-v2/credentials from 1.15.1 to 1.15.2 in
/exporter/awskinesisexporter
Bump github.com/aws/aws-sdk-go-v2/credentials from 1.15.1 to 1.15.2 in
/extension/sigv4authextension
Bump github.com/aws/aws-sdk-go-v2/service/kinesis from 1.22.0 to 1.22.1
in /exporter/awskinesisexporter
Bump github.com/aws/aws-sdk-go-v2/service/sts from 1.25.0 to 1.25.1 in
/exporter/awskinesisexporter
Bump github.com/aws/aws-sdk-go-v2/service/sts from 1.25.0 to 1.25.1 in
/extension/sigv4authextension
Bump github.com/grpc-ecosystem/grpc-gateway/v2 from 2.18.0 to 2.18.1 in
/receiver/opencensusreceiver
Bump github.com/jcchavezs/porto from 0.5.1 to 0.6.0 in /internal/tools
Bump github.com/sijms/go-ora/v2 from 2.7.19 to 2.7.20 in
/receiver/sqlqueryreceiver
Bump github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/common
from 1.0.786 to 1.0.791 in /exporter/tencentcloudlogserviceexporter
Bump go.mongodb.org/mongo-driver from 1.12.1 to 1.13.0 in
/receiver/mongodbreceiver
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /cmd/configschema
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /cmd/otelcontribcol
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /cmd/oteltestbedcol
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /cmd/telemetrygen/internal/e2etest
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /exporter/coralogixexporter
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /exporter/datadogexporter
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /exporter/datadogexporter/integrationtest
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /exporter/loadbalancingexporter
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /exporter/opencensusexporter
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /exporter/skywalkingexporter
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /extension/jaegerremotesampling
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /processor/k8sattributesprocessor
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /processor/routingprocessor
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /processor/servicegraphprocessor
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /processor/spanmetricsprocessor
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /receiver/jaegerreceiver
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /receiver/jmxreceiver
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /receiver/k8sclusterreceiver
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /receiver/k8sobjectsreceiver
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /receiver/kubeletstatsreceiver
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /receiver/lokireceiver
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /receiver/opencensusreceiver
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /receiver/skywalkingreceiver
Bump
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
from 0.45.0 to 0.46.0 in /testbed
Bump go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp from
0.45.0 to 0.46.0 in /examples/demo/client
Bump golang.org/x/crypto from 0.14.0 to 0.15.0 in
/receiver/podmanreceiver
Bump golang.org/x/net from 0.17.0 to 0.18.0 in
/exporter/azuremonitorexporter
Bump golang.org/x/net from 0.17.0 to 0.18.0 in /internal/aws/awsutil
Bump golang.org/x/net from 0.17.0 to 0.18.0 in /receiver/jaegerreceiver
Bump golang.org/x/oauth2 from 0.13.0 to 0.14.0 in
/exporter/f5cloudexporter
Bump golang.org/x/oauth2 from 0.13.0 to 0.14.0 in
/extension/oauth2clientauthextension
Bump golang.org/x/tools from 0.14.0 to 0.15.0 in /internal/tools

---------

Signed-off-by: Alex Boten <alex@boten.ca>
Co-authored-by: Alex Boten <aboten@lightstep.com>

bmbferreira committed:
fix `delete_matching_keys` examples on docs (#29094)


**Documentation:** minor for fox the `delete_matching_keys` since they
were referring to the `delete_key` function

Co-authored-by: Tyler Helmuth <12352919+TylerHelmuth@users.noreply.github.com>

opentelemetrybot committed:
[chore] dependabot updates Tue Nov 14 04:57:31 UTC 2023 (#29268)

Bump actions/github-script from 6 to 7
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/exporter/awscloudwatchlogsexporter
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/exporter/awsemfexporter
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/exporter/awsxrayexporter
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/exporter/datadogexporter
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/extension/observer/ecsobserver
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/internal/aws/cwlogs
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/internal/aws/proxy
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/internal/aws/xray
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/internal/metadataproviders
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/processor/resourcedetectionprocessor
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/receiver/awscontainerinsightreceiver
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/receiver/awsecscontainermetricsreceiver
Bump github.com/aws/aws-sdk-go from 1.47.6 to 1.47.10 in
/receiver/awsxrayreceiver
Bump github.com/aws/aws-sdk-go from 1.47.9 to 1.47.10 in
/internal/aws/awsutil
Bump github.com/aws/aws-sdk-go from 1.47.9 to 1.47.10 in
/internal/aws/k8s
Bump github.com/aws/aws-sdk-go from 1.47.9 to 1.47.10 in
/internal/aws/xray/testdata/sampleapp
Bump github.com/jcchavezs/porto from 0.5.1 to 0.6.0 in /internal/tools
Bump github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/common
from 1.0.786 to 1.0.792 in /exporter/tencentcloudlogserviceexporter

etiennep committed:
[receiver/kubeletstatsreceiver] address lint issues: Implicit Memory Aliasing (#29113)

This PR fixes an implicit memory aliasing lint issue in
the kubeletstatsreceiver

* fixed code
* removed linting exception

https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/28584

Co-authored-by: Tyler Helmuth <12352919+TylerHelmuth@users.noreply.github.com>

rajkumar-rangaraj committed:
[chore] [exporter/azuremonitor] Incorporate SDK Version Tagging in Azure Monitor Exporter (#28999)

**Description:** <Describe what has changed.>
<!--Ex. Fixing a bug - Describe the bug and how this fixes the issue.
Ex. Adding a feature - Explain what this achieves.-->

This pull request enhances the Azure Monitor Exporter by tagging
telemetry data with the OpenTelemetry Collector's internal SDK version.
This additional metadata will facilitate improved monitoring,
troubleshooting, and analysis of data sourced from different versions of
the collector.

**Changes:**
- Added functionality in `azuremonitorexporter` to append the SDK
version to each telemetry envelope.
- Implemented `getCollectorVersion` to dynamically fetch the current
version of the OpenTelemetry Collector.

**Sample SDK Version String:**
- The SDK version string is formatted as follows:
`otelc-<version>-<os>-<arch>`
- Example: `otelc-v0.88.1-lin-amd64`

**Note:**
- Since this change pertains to internal diagnostic data enrichment and
does not modify any external behavior or configuration interfaces, it
does not necessitate a changelog entry.

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were
added.>
- Adjusted existing unit tests to ensure the SDK version string is
correctly included in the telemetry envelopes.

```json
{
	"ver": 1,
	"name": "Microsoft.ApplicationInsights.RemoteDependency",
	"time": "2023-11-06T15:21:12.9807976-08:00",
	"sampleRate": 100,
	"seq": "",
	"iKey": "Ikey",
	"tags": {
		"ai.cloud.role": "otlp-test",
		"ai.cloud.roleInstance": "2c4a6d1b-f7cb-4579-8d05-65b83b4acf7e",
		*"ai.internal.sdkVersion": "otelc-v0.88.1-lin-amd64",*
		"ai.operation.id": "0e3a010cb64d919aa8f214bf0a166334",
		"ai.operation.parentId": ""
	},
	"data": {
	}
}
```

---------

Co-authored-by: Yang Song <songy23@users.noreply.github.com>

gebn committed:
[kafka] Expose resolve_canonical_bootstrap_servers_only (#26022)

**Description:** The latest release of IBM/sarama includes the
`resolve_canonical_bootstrap_servers_only` setting for
`client.dns.lookup`. In a GSSAPI context, this enables replacing a list
of brokers with a single name that resolves to all of the underlying
IPs. Previously this would fail with a Kerberos error. This PR exposes
the new option in collector config.

**Testing:** Added new cases to check the option is set correctly.

**Documentation:** Added a description of the parameter to the relevant
READMEs.

crobert-1 committed:
[chore][exporter/signalfx] Update test certs (#29109)

**Description:** <Describe what has changed.>
<!--Ex. Fixing a bug - Describe the bug and how this fixes the issue.
Ex. Adding a feature - Explain what this achieves.-->
This re-enables the failing tests and updates the expired certificates.

New cert expiration dates:
```
$ openssl x509 -enddate -noout -in testdata/certs/cert.pem 
notAfter=Mar 13 22:55:27 3023 GMT
$ openssl x509 -enddate -noout -in testdata/certs/ca.pem 
notAfter=Mar 13 22:47:43 3023 GMT
```
**Link to tracking Issue:** <Issue number if applicable>
Fixes #29099 

**Testing:** <Describe what testing was performed and which tests were
added.>
Tests are now passing.

**Documentation:** <Describe the documentation added.>

sakulali committed:
[chore][receiver/journald] remove duplicate function NewFactory and pass checkapi (#29005)

**Description:**
Remove duplicate function NewFactory and pass checkapi since we should
only keep one NewFactory.

**Link to tracking Issue:**

https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/26304

**Testing:**
go run cmd/checkapi/main.go .
go test for journaldreceiver

**Documentation:**

Signed-off-by: sakulali <sakulali@126.com>

sigilioso committed:
[receiver/redis] include server.address and server.port resource attributes (#26707)

**Description:** <Describe what has changed.>

Two new resource attributes have been added: `server.address` and
`server.port`. This is change aligned with [semantic
conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.25.0/specification/semantic-conventions.md?plain=1#L30)
and allows users identifying a particular redis server.


**Link to tracking Issue:** #22044

**Testing:** <Describe what testing was performed and which tests were
added.>

```
❯ go test -race -timeout 300s -parallel 4 --tags="",integration --count 1 ./...
ok      github.com/open-telemetry/opentelemetry-collector-contrib/receiver/redisreceiver        6.179s
ok      github.com/open-telemetry/opentelemetry-collector-contrib/receiver/redisreceiver/internal/metadata      1.464s
```

* Added: new test to check error when the configuration endpoint is
missing or invalid
* Updated: integration test to check new resource attributes

**Documentation:** <Describe the documentation added.>
New attribute documentation was generated using `mdatagen`

---------

Co-authored-by: Antoine Toulme <antoine@toulme.name>

schmikei committed:
[receiver/mongodbatlas] Logging enhancement for logs retrieval (#29084)

**Description:** Enhances current logs by adding more context to the
warning message and adds a debug log that will help orient users if they
cannot reach that block of code.

**Link to tracking Issue:** Resolves #28851

**Testing:**
```log
2023-11-09T10:11:17.716-0500	warn	mongodbatlasreceiver@v0.88.0/logs.go:245	Attempted to retrieve host logs but received 0 logs	{"kind": "receiver", "name": "mongodbatlas", "data_type": "logs", "error": "EOF", "log": "mongos.gz", "hostname": "cluster1-shard-00-00.t5hdg.mongodb.net", "startTime": "2023-11-09T10:06:16.618-0500", "endTime": "2023-11-09T10:11:16.618-0500"}
2023-11-09T10:11:17.946-0500	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "logging", "resource logs": 1, "log records": 28}
2023-11-09T10:11:18.131-0500	warn	mongodbatlasreceiver@v0.88.0/logs.go:274	Attempted to retrieve audit logs but received 0 logs	{"kind": "receiver", "name": "mongodbatlas", "data_type": "logs", "error": "EOF", "hostname": "cluster1-shard-00-00.t5hdg.mongodb.net", "log": "mongos-audit-log.gz", "startTime": "2023-11-09T10:06:16.618-0500", "endTime": "2023-11-09T10:11:16.618-0500"}
```


**Documentation:** <Describe the documentation added.>

songy23 committed:
[exporter/datadog] Deprecate peer_service_aggregation in favor of peer_tags_aggregation (#29089)

**Description:**
Deprecate peer_service_aggregation in favor of peer_tags_aggregation.
Counterpart of https://github.com/DataDog/datadog-agent/pull/20550.

mrsillydog committed:
[bugfix][apachesparkreceiver] Correct inaccurate metric units (#29104)

**Description:** Replacing inaccurate units for the
spark.job.stage.active and spark.job.stage.result metrics for the Apache
Spark receiver.


**Link to tracking Issue:**
[29104](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/29104)

**Testing:** N/A

**Documentation:** Documentation for the metadata.yaml file was updated
accordingly.

mackjmr committed:
[exporter/datadogexporter] Add support for custom container tags (#29156)

**Description:**
This PR updates `github.com/DataDog/datadog-agent/pkg/trace` and
`github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/attributes`. This
brings in the following PRs:
https://github.com/DataDog/opentelemetry-mapping-go/pull/193,
https://github.com/DataDog/datadog-agent/pull/20779 which add support
for custom container tags (for traces, metrics and logs) via resource
attributes prefixed by `datadog.container.tag.*`.

This PR also brings a couple other changes for traces which are
reflected in the changelog:
- Stop extracting container tags from span attributes, but only extract
from resource attributes. The semantic conventions we are using to
extract the container tags are meant for resource attributes, so we
should only check resource attributes.
- Stop setting the DD Format container tags as span tags, but only set
these in the dedicated container tag (`_dd.container.tag`). We currently
set the DD mapped container tags both as span tag, and in the dedicated
container tag `_dd.tags.container`. This should only be set in the
dedicated container tag.

The update to opentelemetry-mapping-go also brings in this change:
https://github.com/DataDog/opentelemetry-mapping-go/pull/174, which adds
support for `host.cpu.*` attributes.

djaglowski committed:
Fix panic when LogRecordCount is called after ConsumeLogs (#29274)

Resolves #29107
Resolves #27469

renovate-bot committed:
Update github-actions deps (major) (#29244)

[![Mend
Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Type | Update | Change |
|---|---|---|---|
| [actions/github-script](https://togithub.com/actions/github-script) |
action | major | `v6` -> `v7` |

---

### Release Notes

<details>
<summary>actions/github-script (actions/github-script)</summary>

### [`v7`](https://togithub.com/actions/github-script/compare/v6...v7)

[Compare
Source](https://togithub.com/actions/github-script/compare/v6...v7)

</details>

---

### Configuration

📅 **Schedule**: Branch creation - "on tuesday" (UTC), Automerge - At any
time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

👻 **Immortal**: This PR will be recreated if closed unmerged. Get
[config help](https://togithub.com/renovatebot/renovate/discussions) if
that's undesired.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR has been generated by [Mend
Renovate](https://www.mend.io/free-developer-tools/renovate/). View
repository job log
[here](https://developer.mend.io/github/open-telemetry/opentelemetry-collector-contrib).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNy40Ni4wIiwidXBkYXRlZEluVmVyIjoiMzcuNDYuMCIsInRhcmdldEJyYW5jaCI6Im1haW4ifQ==-->

---------

Co-authored-by: Alex Boten <alex@boten.ca>

diurnalist committed:
make distribution and histogram distinct types

there is a subtle difference between the Histogram and Distribution
type; the Distribution is specific to DogStatsD and performs global
aggregation by taking in raw values. the Histogram performs local
(node) aggregation and outputs preaggregated avg/count/max/median
and optionally percentile aggregations.

this updates the receiver to treat these as explicit types, which must
be configured separately. prior, there was behavior where the same
config would be applied to each statsd type even if the end-user desired
them to differ.

diurnalist committed:
use quantiles instead of adhoc aggregation

also corrects some bad math
